platform_buckets_api/service.py

```diff
@@ -135,15 +135,20 @@ async def get_bucket(self, id: str) -> BaseBucket:
         return await self._storage.get_bucket(id)
 
     async def get_bucket_by_name(
-        self,
-        name: str,
-        org_name: Optional[str] = None,
-        project_name: Optional[str] = None,
-        owner: Optional[str] = None,
+        self, name: str, org_name: Optional[str], project_name: str
     ) -> BaseBucket:
-        return await self._storage.get_bucket_by_name(
-            name=name, org_name=org_name, project_name=project_name, owner=owner
-        )
+        try:
+            return await self._storage.get_bucket_by_name(name, org_name, project_name)
+        except NotExistsError:
+            # project_name could be a username if user is searching for
+            # legacy bucket which does't have project. We need to search
+            # for user bucket also. It is important to check that it is a legacy bucket
+            # before returning it since legacy buckets and project buckets
+            # without org have the same name format.
+            bucket = await self._storage.get_bucket_by_name(name, None, project_name)
```

Comment:
So, in this case, we still need to pass `owner=project_name`, don't we?

Reply:
No, I removed this parameter, it's not needed. Legacy buckets have project_name = owner, you just need to check that it's a legacy bucket afterwards.

Reply:
As I understand, `PROJECT_LABEL` was added only in https://github.com/neuro-inc/platform-buckets-api/pull/398, so buckets created prior to 398's deployment wouldn't have project_name. Or did we add some migration for buckets too, not sure, maybe I'm missing something.

Reply:
No, there is no migration. This case is handled and also covered by tests. When there is no project label owner is used as a project_name https://github.com/neuro-inc/platform-buckets-api/blob/master/platform_buckets_api/kube_client.py#L125.
Project label is used only to store project name, it's not used in buckets filtering.